.github/workflows/deploy.yml

```diff
@@ -354,3 +366,76 @@ jobs:
         with:
           name: cypress-videos
           path: tests/cypress/videos
+      - name: upload partial cypress coverage
+        uses: actions/upload-artifact@v3
+        with:
+          name: partial-cypress-coverage-${{ matrix.containers}}
+          path: ./coverage-cypress/lcov.info
+
+      - name: upload partial cypress coverage json
+        uses: actions/upload-artifact@v3
+        with:
+          name: cypress-json-coverage-${{ matrix.containers}}
+          path: ./coverage-cypress/coverage-final.json
+
+      - name: publish code coverage
+        uses: paambaati/codeclimate-action@v3.0.0
+        continue-on-error: true
+        env:
+          CC_TEST_REPORTER_ID: cace182021fe88d327fa8d95355ac6081f420e094a214fe77feb5df2f0259e9d
+        with:
+          debug: true
+          coverageLocations: |
+            ${{github.workspace}}/coverage-cypress/lcov.info
+  coverage:
+    name: test-coverage
+    needs: [cypress]
+    if: always() && needs.finishing-prep.result == 'success'
+    runs-on: ubuntu-latest
+    permissions:
+      deployments: write
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Collate test coverage reports
+        run: |
+          ls
+          echo $(pwd)
+          cd ./services/app-web
+          echo $(pwd)
+          ls
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+
+      - name: Create combined test coverage report
+        run: |
+          echo "Combining coverage reports"
+          mkdir -p reports
+          mkdir -p cypress_temp
+          mkdir -p .nyc_output
+          echo "listing current folder"
+          ls
+          echo "listing a cypress folder"
+          ls ./cypress-json-coverage-1/
+          cp ./cypress-json-coverage-1/coverage-final.json ./cypress_temp/from-cypress-1.json
+          cp ./cypress-json-coverage-2/coverage-final.json ./cypress_temp/from-cypress-2.json
+          cp ./cypress-json-coverage-3/coverage-final.json ./cypress_temp/from-cypress-3.json
+          cp ./cypress-json-coverage-4/coverage-final.json ./cypress_temp/from-cypress-4.json
```

Comment:
This a pretty long script. Can we quickly document whats is happening somewhere - maybe make a new file `docs/testCoverage` with a couple sentences about this approach.

Reply:
added some echo lines to make it clear what's happening.  it's really just a bunch of copying, and a couple of merges...